projects/driver_controls/test/test_throttle.c

```diff
@@ -0,0 +1,162 @@
+#include <string.h>
+#include "ads1015.h"
+#include "ads1015_def.h"
+#include "delay.h"
+#include "gpio_it.h"
+#include "input_event.h"
+#include "interrupt.h"
+#include "throttle.h"
+#include "unity.h"
+
+static Ads1015Storage ads1015_storage;
+static ThrottleStorage throttle_storage;
+static ThrottleCalibrationData calibration_data;
+static int16_t s_mocked_reading;
+static EventID s_mocked_event;
+static int16_t s_threshes_main[NUM_THROTTLE_ZONES][NUM_THROTTLE_THRESHES] = {
+  { 325, 625 },   //
+  { 625, 1135 },  //
+  { 1135, 1404 }  //
+};
+static int16_t s_threshes_secondary[NUM_THROTTLE_ZONES][NUM_THROTTLE_THRESHES] = {
+  { 162, 312 },  //
+  { 312, 575 },  //
+  { 575, 707 }   //
+};
+static int16_t s_tolerance = 10;
+
+StatusCode TEST_MOCK(ads1015_read_raw)(Ads1015Storage *storage, Ads1015Channel channel,
+                                       int16_t *reading) {
+  if (channel == throttle_storage.channel_main) {
+    *reading = s_mocked_reading;
+  } else {
+    *reading = s_mocked_reading / 2;
+  }
+  return STATUS_CODE_OK;
+}
+
+StatusCode TEST_MOCK(event_raise)(EventID id, uint16_t data) {
```

Comment:
Why are you mocking `event_raise`? Couldn't you just pop the event?

I think it's better to avoid mocking where we can so we're actually testing our code and not just throwaway code.